tests/st/calicoctl/test_default_pools.py

```diff
@@ -0,0 +1,162 @@
+        (True, "CALICO_IPV4POOL_CIDR", "10.0.0.0/24", 2, None, "Typical non-default pool"),
+        (True, "CALICO_IPV4POOL_CIDR", "10.0.0.0/26", 2, None, "Smallest legal pool"),
+        (True, "CALICO_IPV6POOL_CIDR", "fd00::/122", 2, None, "Smallest legal pool"),
+        (False, "CALICO_IPV6POOL_CIDR", "fd00::/123", 0, None, "Too small"),
+        (False, "CALICO_IPV6POOL_CIDR", "fd00::/128", 0, None, "Too small, but legal CIDR"),
+        (False, "CALICO_IPV6POOL_CIDR", "fd00::/129", 0, None, "Impossible CIDR"),
+        (True, "CALICO_IPV4POOL_CIDR", "10.0.0.0/24", 2, "cross-subnet", "Typ. non-def pool, IPIP"),
+        (True, "CALICO_IPV4POOL_CIDR", "10.0.0.0/24", 2, "always", "Typ. non-default pool, IPIP"),
+        (True, "CALICO_IPV4POOL_CIDR", "10.0.0.0/24", 2, "off", "Typical pool, explicitly no IPIP"),
+        (True, "CALICO_IPV6POOL_CIDR", "fd00::/122", 2, "always", "IPv6 - IPIP not permitted"),
+        (True, "CALICO_IPV6POOL_CIDR", "fd00::/122", 2, "cross-subnet", "IPv6 - IPIP not allowed"),
+        (True, "CALICO_IPV6POOL_CIDR", "fd00::/122", 2, "off", "IPv6, IPIP explicitly off"),
+        (False, "CALICO_IPV6POOL_CIDR", "fd00::/122", 0, "junk", "Invalid IPIP value"),
+        (False, "CALICO_IPV4POOL_CIDR", "10.0.0.0/24", 0, "reboot", "Invalid IPIP value"),
+        (True, "CALICO_IPV4POOL_CIDR", "0.0.0.0/0", 2, None, "Valid, but not sensible"),
+        (True, "CALICO_IPV6POOL_CIDR", "::/0", 2, None, "Valid, but not sensible"),
+        (True, "CALICO_IPV6POOL_CIDR", "fd80::0:0/120", 2, None, "Valid, but non-canonical form"),
+        (False, "CALICO_IPV6POOL_CIDR", "1.2.3.4/24", 0, None, "Wrong type"),
+        (False, "CALICO_IPV4POOL_CIDR", "fd00::/24", 0, None, "Wrong type"),
+        (True, "CALICO_IPV6POOL_CIDR", "::0:a:b:c:d:e:0/120", 2, None, "Valid, non-canonical form"),
+        (False, "CALICO_IPV4POOL_CIDR", "1.2/16", 0, None, "Valid, unusual form"),
+    ])
+    def test_default_pools(self, success_expected, param, value, exp_num_pools, ipip, description):
+        """
+        Test that the various options for default pools work correctly
+        """
+        _log.debug("Test description: %s", description)
+        # Get command line for starting docker
+        output = self.host.calicoctl("node run --dryrun --node-image=%s" % NODE_CONTAINER_NAME)
+        base_command = output.split('\n')[-4].rstrip()
+
+        # Modify command line to add the options we want to test
+        env_inserts = "-e %s=%s " % (param, value)
+        if ipip is not None:
+            env_inserts += "-e CALICO_IPV4POOL_IPIP=%s " % ipip
+        prefix, _, suffix = base_command.partition("-e")
+        command = prefix + env_inserts + "-e" + suffix
+
+        # Start calico-docker
+        self.host.execute(command)
+
+        if not success_expected:
+            # check for "Calico node failed to start"
+            expected_string = "Calico node failed to start"
```

Comment:
I'd probably wrap this up as `wait_for_node_log("Callico node failed to start")`, for example